src/sentry/integrations/issue_alert_image_builder.py

```diff
@@ -32,33 +43,51 @@ def __init__(self, group: Group, provider: ExternalProviderEnum) -> None:
             "provider": self.provider,
             "issue_category": self.group.issue_category,
         }
+        self.lock = locks.get(key=f"lock_{self.cache_key}", duration=10, name="issue_alert_image")
 
     def get_image_url(self) -> str | None:
         try:
-            image_url = None
             metrics.incr("chartcuterie.issue_alert.attempt", tags=self.tags)
-            if self.group.issue_type == PerformanceP95EndpointRegressionGroupType:
-                image_url = self._get_endpoint_regression_image_url()
-            elif self.group.issue_type == ProfileFunctionRegressionType:
-                image_url = self._get_function_regression_image_url()
-
-            if image_url:
-                metrics.incr("chartcuterie.issue_alert.success", tags=self.tags)
-                return image_url
+            image_url = cache.get(self.cache_key)
+            if image_url is None:
+                self.lock.blocking_acquire(initial_delay=1, timeout=30)
+                # Checking again in case another thread generated the image while
+                # this thread was acquiring the lock
+                image_url = cache.get(self.cache_key)
+                if image_url is None:
+                    if self.group.issue_type == PerformanceP95EndpointRegressionGroupType:
+                        image_url = self._get_endpoint_regression_image_url()
+                    elif self.group.issue_type == ProfileFunctionRegressionType:
+                        image_url = self._get_function_regression_image_url()
+        except UnableToAcquireLock:
+            # There is a chance that another thread generated the image
+            image_url = cache.get(self.cache_key)
+            if not image_url:
+                logger.warning(
```

Comment:
I will monitor this and if it happens a lot it means the timeout and acquire numbers should be adjusted. They seem reasonable now. It's a non-breaking thing because worst case the image won't be attached. The alert will still be sent.